glean-core/ffi/src/labeled.rs

```diff
@@ -0,0 +1,93 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use lazy_static::lazy_static;
+use std::convert::TryFrom;
+
+use glean_core::{metrics::*, CommonMetricData, Lifetime};
+
+use super::handlemap_ext::HandleMapExtension;
+use super::*;
+
+macro_rules! impl_labeled_metric {
+    ($metric:ty, $global:ident, $metric_global:ident, $new_name:ident, $destroy:ident, $get_name:ident) => {
+        lazy_static! {
+            static ref $global: ConcurrentHandleMap<LabeledMetric<$metric>> =
+                ConcurrentHandleMap::new();
+        }
+        define_handle_map_deleter!($global, $destroy);
+
+        #[no_mangle]
+        pub extern "C" fn $new_name(
```

Comment:
nit: we should start documenting all public functions. Can you please add a comment for all `pub` functions in this PR?